extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/AccessLogConfig.java

```diff
@@ -26,10 +27,28 @@ public interface AccessLogConfig {
      * - long: `%r\n%{ALL_REQUEST_HEADERS}`
      * <p>
      * Otherwise, consult the Quarkus documentation for the full list of variables that can be used.
+     *
+     * Note that enabling the `%{ALL_REQUEST_HEADERS}` attribute directly or with a `long` named format introduces a risk
+     * of sensitive header values being logged.
+     * <p>
+     * HTTP `Authorization` header value is always masked. Use the {@link #maskedHeaders()} property to mask other sensitive
+     * headers.
      */
     @WithDefault("common")
     String pattern();
 
+    /**
+     * List of HTTP headers whose values must be masked when the `%{ALL_REQUEST_HEADERS}` attribute
+     * is enabled with the {@link #pattern()} property.
+     */
+    Optional<List<String>> maskedHeaders();
```

Comment:
Why don't we define the `Authorization` header as a default value here?

This way, people could potentially override it if absolutely needed?

Reply:
There is a format option there that allows to log specific headers and cookies and indeed I did not update those attributes.

So if the users will really want to see Authorization header values in the logs, they can still choose to log them.

I suppose with the ALL_REQUEST_HEADERS, it is easier for users log some sensitive technical headers without them actually intending to log them, so fot this format option we just mask Authorization.

Does it sound reasonable?

Reply:
@gsmet @mabartos Indeed, I kind of lean toward masking it by default too but without setting `Authorization` as a default masked header as it can lead to users doing more work or omitting it by mistake if they want to mask `Authorization` and some other header. The option to get complete `Authorization` into logs with `%{i,request_header_name}` will remain for users who really need it.